ui/base-tool/src/main/java/org/cru/godtools/base/tool/databinding/PicassoImageViewAdapters.kt

```diff
@@ -1,22 +1,58 @@
 package org.cru.godtools.base.tool.databinding
 
+import android.view.View
 import androidx.databinding.BindingAdapter
+import jp.wasabeef.picasso.transformations.CropTransformation.GravityHorizontal
+import jp.wasabeef.picasso.transformations.CropTransformation.GravityVertical
+import org.ccci.gto.android.common.picasso.view.PicassoImageView
 import org.ccci.gto.android.common.picasso.view.SimplePicassoImageView
-import org.cru.godtools.base.tool.model.view.bindBackgroundImage
-import org.cru.godtools.base.tool.model.view.setPicassoResource
+import org.cru.godtools.base.tool.widget.ScaledPicassoImageView
 import org.cru.godtools.base.tool.widget.SimpleScaledPicassoImageView
+import org.cru.godtools.base.toolFileManager
 import org.cru.godtools.xml.model.ImageGravity
 import org.cru.godtools.xml.model.ImageScaleType
 import org.cru.godtools.xml.model.Resource
+import org.cru.godtools.xml.model.layoutDirection
 
-@BindingAdapter("picassoFile")
+private const val PICASSO_FILE = "picassoFile"
+
+@BindingAdapter(PICASSO_FILE)
 internal fun SimplePicassoImageView.setPicassoFile(resource: Resource?) = setPicassoResource(resource)
 
-@BindingAdapter("picassoFile", "scaleType", "gravity", requireAll = false)
+@BindingAdapter(PICASSO_FILE, "scaleType", "gravity")
 internal fun SimpleScaledPicassoImageView.bindScaledResource(
     resource: Resource?,
     scaleType: ImageScaleType?,
     gravity: Int?
-) = bindBackgroundImage(
-    resource, scaleType ?: ImageScaleType.FIT, gravity?.let { ImageGravity(it) } ?: ImageGravity.CENTER
-)
+) = bindScaledResource(resource, scaleType, gravity?.let { ImageGravity(it) } ?: ImageGravity.CENTER)
+
+private fun PicassoImageView.setPicassoResource(resource: Resource?) =
+    setPicassoFile(resource?.localName?.let { context.toolFileManager.getFileBlocking(it) })
+
+private fun ScaledPicassoImageView.bindScaledResource(
```

Comment:
What's the difference between using ```ScaledPicassoImageView``` and ```SimpleScaledPicassoImageView```? ~~Did we switch to using the non-simple one so we needed to overwrite the values(resource, scaleType, etc)?~~ I realize now you just moved this from the ```ResourceViewsUtil.kt``` file into here.

Reply:
yeah, consolidated code into the binding adapter since that was the only place it was actually getting used. before we used data binding the logic was called from a few different locations.